src/lib.rs

```diff
@@ -301,6 +288,9 @@ pub use crate::util::ptr::Ptr;
 #[allow(unused_imports)]
 use crate::util::polyfills::NonNullExt as _;
 
+#[cfg(feature = "byteorder")]
+compile_error!("the `byteorder` feature has been removed; its functionality is now available regardless of feature set");
```

Comment:
This will cause problems for anyone using `--all-features`, and that includes us: our Kani tests are currently failing for this reason.

One option would be to have this emit a deprecation warning rather than an error. In our CI tests, we could `-Dwarnings` (as we do today) but also add `-Adeprecated` so that we can test any of our own deprecated items such as this one.

Reply:
Discussed offline with @jswrenn. Decided to just remove this feature entirely on the theory that most users aren't relying on it directly (since, on 0.7.X, it's a default feature) and for those that are, it should be obvious what to do.